Sources/Datadog/Tracing/AutoInstrumentation/URLSessionTracingHandler.swift

```diff
@@ -55,20 +55,17 @@ internal class URLSessionTracingHandler: URLSessionInterceptionHandler {
         span.setTag(key: OTTags.httpMethod, value: method)
 
         if let error = resourceCompletion.error {
-            span.setTag(key: OTTags.error, value: true)
-
-            let dderror = DDError(error: error)
-            span.setTag(key: DDTags.errorType, value: dderror.title)
-            span.setTag(key: DDTags.errorMessage, value: dderror.message)
-            span.setTag(key: DDTags.errorStack, value: dderror.details)
+            span.setError(error)
         }
 
-        if let httpResponseStatusCode = resourceCompletion.httpResponse?.statusCode {
-            span.setTag(key: OTTags.httpStatusCode, value: httpResponseStatusCode)
-
-            if httpResponseStatusCode >= 400 && httpResponseStatusCode < 500 {
-                span.setTag(key: OTTags.error, value: true)
-                if httpResponseStatusCode == 404 {
+        if let httpResponse = resourceCompletion.httpResponse {
+            let httpStatusCode = httpResponse.statusCode
+            span.setTag(key: OTTags.httpStatusCode, value: httpStatusCode)
+            if httpStatusCode >= 400 && httpStatusCode < 500 {
+                if let error = httpResponse.asError() {
```

Comment:
400 < status code < 500 logic is in `asError()` method already, so we can remove the first `if` here

Reply:
I thought about that as well but left the outer `if` to keep it explicit what we were checking for. You're right it's extraneous.
I'll rename `asError` to something like `asClientError` to remind of the 400 <= status code < 500 logic.